### ISSUE TITLE
Error compiling with 2.13.0-RC1

### ISSUE THREAD

**Issue by lrytz** [akka/akka-http#2515]
Check out current master (04eb7550c)

In sbt:
```
akka-http-root > set every scalaVersion := "2.13.0-RC1"
akka-http-root > set every scalaBinaryVersion := "2.13.0-M5"
akka-http-root > compile
...
[error] /Users/luc/scala/akka-http/akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala:111:55: diverging implicit expansion for type akka.parboiled2.support.ActionOps.SJoin[akka.shapeless.HNil,Int :: Int :: akka.shapeless.HNil,Int]
[error] starting with method terminate in object Aux
[error]   def date2 = rule { day ~ '-' ~ month ~ '-' ~ digit2 ~> (y ⇒ if (y <= 69) y + 2000 else y + 1900) }
[error]                                                       ^
```

I got the same error on the community build (https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/2018/console).

**Comment by lrytz** [comment]
@raboof / @jrudolph would be really nice if someone could look into this 😊  it's on the path to getting playframework in the community build green.

**Comment by raboof** [comment]
I won't pretend I understand exactly what's going on here, but adding explicit parens seemed to help for me:

    def date2 = rule { day ~ '-' ~ month ~ '-' ~ (digit2 ~> (y ⇒ if (y <= 69) y + 2000 else y + 1900)) }

**Comment by lrytz** [comment]
Thanks, I will use that for now. Leaving this ticket open.

**Comment by lrytz** [comment]
More chnages for RC1 here: https://github.com/scalacommunitybuild/akka-http/commits/community-build-2.13

**Comment by jrudolph** [comment]
> I won't pretend I understand exactly what's going on here, but adding explicit parens seemed to help for me:

Maybe some precedence rules changed? Anyway the extra parens make it more readable anyway.

**Comment by lrytz** [comment]
> Maybe some precedence rules changed?

I think not, with `-Xprint:parser`, the trees are the same in 2.12.8 and 2.13.0-RC1.

**Comment by lrytz** [comment]
Moved here: https://github.com/scala/bug/issues/11516

**Comment by adriaanm** [comment]
Note that this will also hit in 2.12, I think, since the change in behavior landed in 2.13 as a merge from 2.12 (more info in ticket).

**Comment by dwijnand** [comment]
Fixed here in #2525
